### ISSUE TITLE
Allow setting for disk mode sata

### ISSUE THREAD

**Issue by abbbi** [dmacvicar/terraform-provider-libvirt#535]
hi,

currently the plugin only supports two disk types: virtio(-scsi) and virtio.
IDE is used for CDROM.

There are certain systems that dont work with the scsi or virtio option. Some might need SATA
(like ESXi).

There should be the possiblity to specify SATA as disk bus option too.

**Comment by MalloZup** [comment]
@abbbi  for all this extension features, you can use this: https://github.com/dmacvicar/terraform-provider-libvirt/blob/master/website/docs/r/domain.html.markdown#altering-libvirts-generated-domain-xml-definition

:rocket:

**Comment by abbbi** [comment]
@MalloZup 

yes i know, but then it might make sense to have something like:

disk {
  type = "sata|scsi|virtio-scsi|ide"
  ..
}

**Comment by MalloZup** [comment]
yop i get point

**Comment by dmacvicar** [comment]
ℹ️ ℹ️ ℹ️ ℹ️ ℹ️ ℹ️

This issue report is relevant to the legacy version of the provider, which is now in the v0.8 branch of this repository.

Future development is based on a new provider, which is not compatible with this one, nor does share code.

As the new provider solves many issues with the legacy, and to make development in my free time more enjoyable, I have decided to close all bugs for the legacy provider, and to ask to check if the bug would apply to the new one. This also to encourage trying the new version.

- https://registry.terraform.io/providers/dmacvicar/libvirt/latest

and check the documentation:

- https://github.com/dmacvicar/terraform-provider-libvirt/blob/main/README.md
- https://registry.terraform.io/providers/dmacvicar/libvirt/latest/docs

You are free to reopen the bug for v0.8. It may encourage someone to fix it. My efforts will go into the new provider.

I ask you to check the new provider and re-evaluate this bug. 🙏
